docs/runtimes/image-build.md

```diff
@@ -0,0 +1,192 @@
+MLRun generates the dockerfile for the build process, and configures Kaniko with parameters needed for the build.
+
+Building images is done through functions provided by the {py:class}`~mlrun.projects.MlrunProject` class. By using 
+project functions, the same process is used to build and deploy a stand-alone function or functions serving as steps 
+in a pipeline.
+
+## Automatically building images
+MLRun has the capability to auto-detect when a function image needs to first be built. Following is an example that
+will require building of the image:
+
+```python
+project = mlrun.new_project(project_name, "./proj")
+
+project.set_function(
+   "train_code.py", 
+   name="trainer",
+   kind="job",
+   image="mlrun/mlrun",
+   handler="train_func",
+   requirements=["pandas"]
+)
+
+# auto_build will trigger building the image before running, 
+# due to the additional requirements.
+project.run_function("trainer", auto_build=True)
+```
+
+Using the `auto_build` option is only suitable when the build configuration does not change between runs of the
+runtime. For example, if during the development process new requirements were added, the `auto_build` parameter should
+not be used, and manual build will be needed to re-trigger a build of the image.
+
+In the example above, the `requirements` parameter was used to specify a list of additional Python packages required by
+the code. This option directly affects the image build process - each requirement will be installed using `pip` as 
+part of the docker-build process. The `requirements` parameter can also contain a path to a requirements file, making
+it easier to reuse an existing configuration rather than specify a list of packages.
+
+
+## Manually building an image
+
+To manually build an image, use the {py:func}`~mlrun.projects.build_function()` function, which provides multiple 
+options that control and configure the build process.
+
+### Specifying base image
+To use an existing image as the base image for building the image, set the image name in the `base_image` parameter.
+Note that this is the image that will be serving as base (dockerfile `FROM` property), not to be confused with the 
+resulting image name, as specified in the `image` parameter.
+
+```python
+project.build_function(
+   "trainer",
+   base_image="myrepo/my_base_image:latest",
+)
+```
+
+### Running commands
+To run arbitrary commands during the image build, pass them in the `commands` parameter of 
+{py:func}`~mlrun.projects.build_function()`. For example:
+
+```python
+github_repo = "myusername/myrepo.git@mybranch"
+
+project.build_function(
+   "trainer",
+   base_image="myrepo/base_image:latest",
+   commands= [
+        "pip install git+https://github.com/" + github_repo,
+        "mkdir -p /some/path && chmod 0777 /some/path",    
+   ]
+)
+```
+
+These commands will be added as `RUN` operations to the dockerfile generating the image.
```

Comment:
These commands are added as `RUN` operations to the dockerfile generating the image.